torchvision/transforms/v2/functional/_meta.py

```diff
@@ -559,27 +558,23 @@ def _clamp_along_y_axis(
     case_b[..., 6].clamp_(0)  # Clamp x4 to 0
     case_c = torch.zeros_like(case_b)
 
-    cond_a = (x1 < eps) & ~case_a.isnan().any(-1)  # First point is outside left boundary
-    cond_b = y1.isclose(y2, rtol=eps, atol=eps) | y3.isclose(y4, rtol=eps, atol=eps)  # First line is nearly vertical
+    cond_a = (x1 < 0) & ~case_a.isnan().any(-1)  # First point is outside left boundary
+    cond_b = y1.isclose(y2) | y3.isclose(y4)  # First line is nearly vertical
     cond_c = (x1 <= 0) & (x2 <= 0) & (x3 <= 0) & (x4 <= 0)  # All points outside left boundary
     cond_c = (
         cond_c
-        | y1.isclose(y4, rtol=eps, atol=eps)
-        | y2.isclose(y3, rtol=eps, atol=eps)
-        | (cond_b & x1.isclose(x2, rtol=eps, atol=eps))
+        | y1.isclose(y4)
+        | y2.isclose(y3)
+        | (cond_b & x1.isclose(x2))
     )  # First line is nearly horizontal
 
     for (cond, case) in zip(
         [cond_a, cond_b, cond_c],
         [case_a, case_b, case_c],
     ):
         bounding_boxes = torch.where(cond.unsqueeze(1).repeat(1, 8), case.reshape(-1, 8), bounding_boxes)
-    if clamping_mode == "hard":
-        bounding_boxes[..., 0].clamp_(0)  # Clamp x1 to 0
```

Comment:
It's not immediately obvious that this relates to dtypes, so just flagging to make sure this change is intended?

Reply:
@NicolasHug good catch. Yeah this was related to `dtype` and introduction of epsilon. I have remove it in a later commit attached to this PR.